### PR TITLE
soft_999_Jenish_Pate_fw_103

### PR DESCRIPTION
**Pull request by jenishpatel2147**

### CODE REVIEW

File: soft_999_Jenish_Pate_fw_103 [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+#include "interrupt.h"
+#include "log.h"
+#include "soft_timer.h"
+
+#include <stdbool.h>
+#include <stdlib.h>
+
+typedef struct CounterStorage {
+  uint16_t potentiometer_data;   // Stores data
+  AdcChannel address;            // Stores channel
```
Comment (g-jessmuir) [inline soft_999_Jenish_Pate_fw_103:13]: adc channels aren't really addresses, please change it to 'channel' for clarity

File: soft_999_Jenish_Pate_fw_103 [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+} CounterStorage;
+
+ void prv_button_interrupt_handler(const Gpiovoid *empty) {
```
Comment (g-jessmuir) [inline soft_999_Jenish_Pate_fw_103:16]: This isn't the right type for a gpio interrupt callback, please call it something other than empty as well

File: soft_999_Jenish_Pate_fw_103 [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+typedef struct CounterStorage {
+  uint16_t potentiometer_data;   // Stores data
+  AdcChannel address;            // Stores channel
+} CounterStorage;
+
+ void prv_button_interrupt_handler(const Gpiovoid *empty) {    
+      CounterStorage *storage = empty;
+      adc_read_converted(storage->potentiometer_data, &storage->address);
+      LOG_DEBUG("potentiometer data: %d\n", storage->potentiometer_data);
+      delay_ms(200);
```
Comment (g-jessmuir) [inline soft_999_Jenish_Pate_fw_103:20]: please use a #define for values like this. also, why are you delaying here?

File: soft_999_Jenish_Pate_fw_103 [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+}
+
+int main(void) {
+      interrupt_init();       // Initializing Interrupts
```
Comment (g-jessmuir) [inline soft_999_Jenish_Pate_fw_103:24]: formatting issues

File: soft_999_Jenish_Pate_fw_103 [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+      gpio_init();            // Initializing GPIO
+      gpio__it_init():        // Initializing Interrupts
```
Comment (g-jessmuir) [inline soft_999_Jenish_Pate_fw_103:26]: the function is `gpio_it_init()` not `gpio__it_init()`

File: soft_999_Jenish_Pate_fw_103 [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+      LOG_DEBUG("potentiometer data: %d\n", storage->potentiometer_data);
+      delay_ms(200);
+}
+
+int main(void) {
+      interrupt_init();       // Initializing Interrupts
+      gpio_init();            // Initializing GPIO
+      gpio__it_init():        // Initializing Interrupts 
+
+      GpioAddress button = { .port = GPIO_PORT_B, .pin = 2 };                     // Assigning Adress
```
Comment (g-jessmuir) [inline soft_999_Jenish_Pate_fw_103:28]: don't need all the extra space for comments

File: soft_999_Jenish_Pate_fw_103 [hunk 7]
```diff
@@ -0,0 +1,73 @@
...
+
+      GpioAddresss potentiometer_addr = { .port = GPIO_PORT_A, .pin = 6};          // Assigning Adress
+
+      GpioSettings pot_settings {                    // GPIO Settings
```
Comment (g-jessmuir) [inline soft_999_Jenish_Pate_fw_103:32]: formatting issues 🙁

File: soft_999_Jenish_Pate_fw_103 [hunk 8]
```diff
@@ -0,0 +1,73 @@
...
+      GpioAddresss potentiometer_addr = { .port = GPIO_PORT_A, .pin = 6};          // Assigning Adress
+
+      GpioSettings pot_settings {                    // GPIO Settings 
+            GPIO_DIR_IN,
+            GPIO_STATE_LOW,
+            GPIO_RES_NONE,
+            GPIO_ALTFN_ANALOG,
+             };
+
+       GpioSettings s_button_settings {     // GPIO Settings Button Settings (The videos was very helpful)
```
Comment (g-jessmuir) [inline soft_999_Jenish_Pate_fw_103:39]: only use the `s_` prefix if it's actually a static value. Here it's just a variable in a function, so don't use `s_`.

File: soft_999_Jenish_Pate_fw_103 [hunk 9]
```diff
@@ -0,0 +1,73 @@
...
+
+      gpio_init_pin(&potentiometer_addr, &pot_settings);      
+      gpio_init_pin(&button, &s_button_settings);        
+      adc_init(ADC_MODE_SINGLE);
+
+      adc_get_channel(potentiometer_addr, &pot_channel);   // Gets the Address
+      adc_set_channel(pot_channel, true);                  // Enables the Channel
+
+      // Creates the structure
+      CounterStorage Store_in_here = {potentiometer_data = 0, address = pot_channel};
```
Comment (g-jessmuir) [inline soft_999_Jenish_Pate_fw_103:60]: I don't think this is the correct way to assign values in a struct. Also, use `snake_case` for variable names, no capital first letter please

File: soft_999_Jenish_Pate_fw_103 [hunk 10]
```diff
@@ -0,0 +1,73 @@
...
+      // Registers an Interrupt
+      gpio_it_register_interrupt(&potentiometer_addr, &s_interrupt_settings,
+                                 INTERRUPT_EDGE_FALLING, prv_button_interrupt_handler,
+                                 &Store_in_here);               
+
+      while(true) {
+
+      }
+
+ return 0;
```
Comment (g-jessmuir) [inline soft_999_Jenish_Pate_fw_103:71]: formatting issues

**Comment by jenishpatel2147** [comment]
No sorry I didn't build this first time sorry. I'll always remember to build for now on 😅. I want to ask what do you exactly mean by the makefile, do you mean the ###rules.mk file that is created when I created a new project?

**Comment by g-jessmuir** [comment]
> No sorry I didn't build this first time sorry. I'll always remember to build from now on 😅. I want to ask what do you exactly mean by the makefile, do you mean the ###rules.mk file that is created when I created a new project?

Yes, you need to include it so it actually compiles upon submission to github

**Pull request closed without merging**